source/braille.py

```diff
@@ -2472,29 +2480,46 @@ def handleCaretMove(
 		region = self.mainBuffer.regions[-1] if self.mainBuffer.regions else None
 		if region and region.obj==obj:
 			region.pendingCaretUpdate=True
+			self._regionsPendingUpdate.add(region)
 		elif prevTether == TetherTo.REVIEW.value:
 			# The caret moved in a different object than the review position.
 			self._doNewObject(getFocusRegions(obj, review=False))
 
-	def handlePendingCaretUpdate(self):
-		"""Checks to see if the final text region needs its caret updated and if so calls _doCursorMove for the region."""
-		region=self.mainBuffer.regions[-1] if self.mainBuffer.regions else None
-		if isinstance(region,TextInfoRegion) and region.pendingCaretUpdate:
-			try:
-				self._doCursorMove(region)
-			finally:
-				region.pendingCaretUpdate=False
+	if version_year < 2024 and NVDAState._allowDeprecatedAPI():
+		def handlePendingCaretUpdate(self):

```

Comment:
is this change going to materially break any add-ons?
I know we have created no-op aliases in the past, but this seems riskier than other instances.

Reply:
I'd be surprised if there is an add-on that uses this, though we cannot rule that out. However, I don't think it can do any harm. The only way I could think of for an add-on to use this is by explicitly updating the caret (i.e. calling handleCaretMove and handlePendingCaretUpdate in one wave). That would still behave the same as before.